prov/verbs/src/verbs_msg_ep.c

```diff
@@ -221,7 +225,18 @@ static int fi_ibv_msg_ep_enable(struct fid_ep *ep)
 
 	attr.cap.max_inline_data = _ep->info->tx_attr->inject_size;
 
-	attr.srq = NULL;
+	if (_ep->srq_ep) {
+		attr.srq =_ep->srq_ep->srq;
+		/* Use of SRQ, no need to allocate recv_wr entries in the QP */
+		attr.cap.max_recv_wr = 0;
+
+		/* Cannot post a recv WR to a QP where a SRQ is attached to */
+		verbs_ops_msg = fi_ibv_msg_ep_ops_msg(_ep);
+		verbs_ops_msg->recv = fi_no_msg_recv;
+		verbs_ops_msg->recvv = fi_no_msg_recvv;
+		verbs_ops_msg->recvmsg = fi_no_msg_recvmsg;
```

Comment:
The call to fi_ibv_msg_ep_ops_msg is returning a pointer to a static structure.  You can't change the values that it points to.  This ends up changing the receive functions for all message endpoints, regardless of whether they use SRX or not.  Either you need to define a new structure that sets these calls to ENOSYS versions, add checks to the receive calls, or add assertions to the receive calls.